### ISSUE TITLE
Add back/forward buttons. 

### ISSUE THREAD

**Issue by octalmage** [octalmage/minum#3]
Need to add back/forward buttons, or hotkeys, but keep the interface clean.

**Comment by octalmage** [comment]
@therealtaryn and I were talking about this and we thought putting them in the bottom left and right corners might be cool. They would be hidden until you hover over them.

**Comment by octalmage** [comment]
@mjchamplin what are your thoughts on this? I want to avoid putting buttons in the title bar if possible. I'm probably going to put a menu button on the right side though.

**Comment by octalmage** [comment]
One of these:

<img src="http://iqcontent.com/blog/files/wp/2014/03/menu-alt-512.png" width="100">

**Comment by octalmage** [comment]
Here's an idea: 

![](https://cloudup.com/cJeN9UgFE_r+)

Put the back and forward buttons next to the close buttons. They can change colors like the title:

![](https://cldup.com/gnLUw7-ctY.png)     
![](https://cldup.com/jKiXnAfVOy.png)

Thoughts? 
cc: @mjchamplin @Therealtaryn

**Comment by mjchamplin** [comment]
I think this is interesting. I'll mock something up in Sketch tonight.

**Comment by octalmage** [comment]
Yeah it's closer to the standard browser format, but I'm not sold. I still like the idea of using the bottom left and right corners, no clue if it will work though.

**Comment by mjchamplin** [comment]
Yeah I'm conflicted. Being able to use gestures on OSX is awesome, but some people don't. There needs to be a fallback that's good UX practice but isn't a dominant or cluttering visual element. Putting it in the title bar and making it that small is probably going to seem kind of awkward, but adding a toolbar makes it just like every other browser. 

Some variation of the original floating idea would likely be the best compromise. I'm sketching some stuff up, I'll keep you posted.

**Comment by mjchamplin** [comment]
Okay so I've played with the < > in the title bar. I don't like it. 

I've played with the corners, but on a large screen it's weird. They're sort of at the opposite place from what you expect because the back button is ALWAYS in the upper-left. 

I took a break, and was playing on my phone. I use Chrome. They've done this thing now where scrolling above the page pulls down a menu with a refresh button in the center, plus a 'new tab' and 'close tab' button. You only see this interface if you go to the top of the page and then pull down. 

I think this kind of thing would integrate on desktop. Scroll up past the top of the page, and a really simple toolbar pulls down with a back and forward button, geographically where people expect them but hidden until they're needed. This would also look cool with the disappearing title bar, because one would appear when you started scrolling upward, and the other would slide down when you reach the top. 

Does this make sense? I really want to mock this up, but I have to pick my girlfriend up from the airport in a little bit. If I don't get to it tonight I will soon, because I'm stoked.

**Comment by octalmage** [comment]
Hmm my Chrome only has the one address bar. But I'm worried an extra bar might look odd with only a few buttons. 

I'm also worried about the implementation, hiding the titlebar is going to be pretty difficult as is. 

BUT, please mock it up! If it's awesome, I'll totally make it happen.

**Comment by mjchamplin** [comment]
That makes sense -- it being difficult to implement. I'm still working on it / thinking about it. I agree that there might not be enough buttons to warrant an entire bar. I know there's a solution, though!

**Comment by mjchamplin** [comment]
## Totally Rough Mockup.

![normal](https://cloud.githubusercontent.com/assets/1672613/8294675/96940d06-1903-11e5-9eb6-510596e265fd.png)
Normal view
![scrolled](https://cloud.githubusercontent.com/assets/1672613/8294683/a174cb20-1903-11e5-8a02-76b26720ffbe.png)
Scroll up past top of page to reveal toolbar.

**Comment by octalmage** [comment]
Hmm that might be cool. I like having the second page though. We have a lot of potential there. This is so hard. 

I need to finish #7 first, this will make the browser useable. Then I'll play with the UI.

**Comment by octalmage** [comment]
@mjchamplin The more I think about this, the more I realize that we can't have any actual buttons on the screen. Which is why your above idea make a ton of sense. 

How would you feel about the Mac styled gesture based back and forward?

**Comment by mjchamplin** [comment]
I think gesture-based is a no-brainer if it isn't a pain in the ass to impliment. 

Of course discoverability is an issue, not to mention a user who 1) has that feature turned off or 2) is using a mouse without a back button. 

I will keep wishing there was a way to add in a button, though I agree that having a button displayed by default runs counter to the spirit of the project. Something that revealed on hover, maybe, but that still feels a bit too normal-browser.

**Comment by octalmage** [comment]
I’ve seen apps detect when a screenshot is being taken to hide DRM content. It would be crazy if we could do that with the UI. That’s probably insanely complicated and doesn’t cover every use case. Just an idea. I’ve always been a fan of on hover actions like Marknote.

**Comment by mjchamplin** [comment]
I have an idea. I will play around with it and let you know if I come up with anything.
